api/ipam/v1beta2/ipaddress_types.go

```diff
@@ -48,6 +47,43 @@ type IPAddressSpec struct {
 	Gateway string `json:"gateway,omitempty"`
 }
 
+// IPAddressClaimReference is a reference to an IPAddressClaim.
+type IPAddressClaimReference struct {
+	// name of the IPAddressClaim.
+	// name must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	Name string `json:"name"`
+}
+
+// IPPoolReference is a reference to an IPPool.
+type IPPoolReference struct {
+	// name of the IPPool.
+	// name must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	Name string `json:"name"`
+
+	// kind of the IPPool.
+	// kind must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
+	Kind string `json:"kind"`
+
+	// apiGroup of the IPPool.
+	// +required
```

Comment:
This was optional before. I don't know what the intent was though

Reply:
Same for Name in the other structs, but there it definitely makes sense to be required?

Reply:
TBH I don't think that a ref makes sense without Name and API Group.
CC @schrej